### PR TITLE
feat: `--annotation` for `index create`

### PR DESCRIPTION
**Pull request by wangxiaoxuan273**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Part of #1053 

**Please check the following list**:
- [x]  Does the affected code have corresponding tests, e.g. unit test, E2E test?
- [ ]  Does this change require a documentation update?
- [ ]  Does this introduce breaking changes that would require an announcement or bumping the major version?
- [x]  Do all new files have an appropriate license header?

### CODE REVIEW

File: cmd/oras/root/manifest/index/create.go [hunk 1]
```diff
@@ -204,3 +216,18 @@ func pushIndex(ctx context.Context, target oras.Target, desc ocispec.Descriptor,
 	}
 	return printer.Println("Digest:", desc.Digest)
 }
+
+func parseAnnotations(input []string) (map[string]string, error) {
```
Comment (wangxiaoxuan273) [inline cmd/oras/root/manifest/index/create.go:220]: This part is copied and modified from `packer.go`, refactoring may be needed.

Comment (Wwwsylvia) [reply cmd/oras/root/manifest/index/create.go:220]: Do we consider exporting the original `parseAnnotationFlags` function in `packer.go` and reusing it here?

Comment (qweeah) [reply cmd/oras/root/manifest/index/create.go:220]: Better refactor `packer` option to isolate out `annotation` flag parsing

```mermaid
flowchart LR
    subgraph Command
	A(push)
	B(attach)
	C(manifest index create)
    end
    subgraph Option
	subgraph packer
		D[annotation]
	end
    end
    A ----> packer
    B ----> packer
    C --> D
```

You man also replace`LoadManifestAnnotations()` with `annotation.Parse`

**Review by Wwwsylvia — APPROVED** [review]
LGTM. But I am not a maintainer.

File: cmd/oras/internal/option/packer.go [hunk 2]
```diff
@@ -91,27 +91,24 @@ func (opts *Packer) Parse(*cobra.Command) error {
...
-		return nil, errAnnotationConflict
+// parseAnnotations loads the manifest annotation map.
+func (opts *Packer) parseAnnotations(cmd *cobra.Command) (err error) {
+	if opts.AnnotationFilePath != "" && len(opts.ManifestAnnotationFlags) != 0 {
+		return errAnnotationConflict
 	}
 	if opts.AnnotationFilePath != "" {
-		if err = decodeJSON(opts.AnnotationFilePath, &annotations); err != nil {
-			return nil, &oerrors.Error{
+		if err = decodeJSON(opts.AnnotationFilePath, &opts.Annotations); err != nil {
```
Comment (shizhMSFT) [inline cmd/oras/internal/option/packer.go:103]: `opts.Annotations` set by `opts.AnnotationFilePath` will be overwritten by `opts.ManifestAnnotationFlags` as per `annotation.go:45`.

**Review by shizhMSFT — CHANGES_REQUESTED** [review]

File: cmd/oras/internal/option/packer.go (hunk 2 above)
Comment (shizhMSFT) [reply cmd/oras/internal/option/packer.go:103]: Those flags cannot be used together.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Review by shizhMSFT — APPROVED** [review]
LGTM

File: cmd/oras/internal/option/annotation.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+	// Annotations contains parsed manifest and config annotations
+	Annotations map[string]map[string]string
+}
+
+// ApplyFlags applies flags to a command flag set.
+func (opts *Annotation) ApplyFlags(fs *pflag.FlagSet) {
+	fs.StringArrayVarP(&opts.ManifestAnnotations, "annotation", "a", nil, "manifest annotations")
+}
+
+func (opts *Annotation) Parse(*cobra.Command) error {
```
Comment (qweeah) [inline cmd/oras/internal/option/annotation.go:47]: Missing doc comment

File: cmd/oras/internal/option/annotation.go [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+	"github.com/spf13/pflag"
+	oerrors "oras.land/oras/cmd/oras/internal/errors"
+)
+
+var (
+	errAnnotationFormat      = errors.New("annotation value doesn't match the required format")
+	errAnnotationDuplication = errors.New("duplicate annotation key")
+)
+
+// Packer option struct.
```
[the remark this answers is not in the document]
Comment (wangxiaoxuan273) [reply cmd/oras/internal/option/annotation.go:33]: Thanks for catching

**Pull request merged**